src/browser/replay/recorder.js

```diff
@@ -4,15 +4,24 @@ import { EventType } from '@rrweb/types';
 import hrtime from '../../tracing/hrtime.js';
 import logger from '../../logger.js';
 
+/** @typedef {import('./recorder.js').BufferCursor} BufferCursor */
+
 export default class Recorder {
   _options;
   _rrwebOptions;
+
+  _isReady = false;
   _stopFn = null;
   _recordFn;
-  _events = {
-    previous: [],
-    current: [],
-  };
+
+  /** A two-slot ring buffer for storing events. */
+  _buffers = [[], []];
+  /** Active slot index (0|1). Stores new events until next checkout. */
+  _currentSlot = 0;
+  /** Index of the finalized inactive slot (0|1). Frozen until next checkout. */
+  get _previousSlot() {
+    return this._currentSlot ^ 1;
```

Comment:
Since the ring has two slots we can just XOR which effectively flips the bit, so we avoid the modulo from `(currentIndex + 1) % 2`.